### PR TITLE
Allow customizing/overwriting the file entries display lines

### PR DESCRIPTION
**Pull request by ryanoasis**
* adds optional customizable global variable
* updates documentation with the option and an example usage

created for: https://github.com/ryanoasis/vim-devicons/issues/94

my example use case:

```vim
let g:startify_custom_entry_display = "\" (\" . WebDevIconsGetFileTypeSymbol(entry_path) . \") \" . repeat(' ', (3 - strlen(index))) . entry_path"
```

selection_19_08_15_19 45 59_813x645_001

* one thing I noticed was syntax highlight was a bit off but maybe that is okay? (would only affect those using vim-devicons or those making use of this new global)

Thanks!

### CODE REVIEW

**Comment by ryanoasis** [comment]
Whoops just noticed I forgot to update the text in the documentation. Fail :disappointed:

**Comment by mhinz** [comment]
The only problem I currently see is highlighting. Everything after the index `[x]  ` is supposed to be a path and thus highlighted like that. Adding in front of the entry will look different than appending etc.

If can't come up with a better approach, I'll merge this in ~2 weeks. I'm about to leave for a trip to Erland land, ehm, Sweden.

**Comment by ryanoasis** [comment]
Thanks for the quick reply! :smile: 

Understood, let me know if I should update the doc that has copied pasted text and add another commit or resubmit a PR.

Oh and have a safe journey

**Comment by ryanoasis** [comment]
Friendly ping. Should I correct the doc that I copy-pasted as well?

**Comment by mhinz** [comment]
Hey, I was longer away than planned.

Usually I'm in favor of approaches that make it easy to customize stuff, but in this case, I'd like to avoid another new option and simply hardcode vim-devicons support. There's just too much highlighting going on that can break and honestly I'd like to avoid any follow-up issues because of that. (Proper highlighting will always be my biggest issue with Vim.)

Basically the plugin would do `if exists('WebDevIconsGetFileTypeSymbol') ...`.

Would you mind this change?

**Comment by ryanoasis** [comment]
Hey no problem really. :smile: 

Hardcoding the support is fine with me, I understand your reasoning behind it.

I might (at a later date) be renaming some API stuff and public functions to not be as horribly named in vim-devicons :stuck_out_tongue: but for now the checking of `WebDevIconsGetFileTypeSymbol` would be fine and even later it would be a simple update.

So no I would *not* mind the change you proposed. :+1: Thanks.

**Comment by mhinz** [comment]
Since I'm using Vim on OSX right now, I don't have a GUI.

Could you please check if this branch works for you?

```
git pull && git checkout devicons
```

**Comment by ryanoasis** [comment]
@mhinz Thanks, I will try to check it this week or this weekend :+1:

**Comment by ryanoasis** [comment]
Looks good to me, here is what I tested (in gvim and vim on Linux):

* entering the file number to go to the file
* double clicking to go to the file (gui)
* launching Startify in command line mode

**Comment by mhinz** [comment]
Well, if the icons are properly aligned, then I'll merge it later today.

Thanks for testing.

**Comment by mhinz** [comment]
Merged! 

Thanks for this innovative plugin. :-)

**Comment by ryanoasis** [comment]
Great thanks! No problem it is enjoyable (mostly). I use startify now as well. Cheers
